flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -472,7 +472,9 @@ public static String getStartCommand(String template,
 		for (Map.Entry<String, String> variable : startCommandValues
 			.entrySet()) {
 			template = template
-				.replace("%" + variable.getKey() + "%", variable.getValue());
+				.replace("%" + variable.getKey() + "%", variable.getValue())
```

Comment:
I think for the test cases that verifies the generated shell command, the main purpose is to verified the command values and their order. How many spaces between tokens is rather an unconcerned implementation detail.
Previously, we need to carefully put one or multiple sequential spaces between the command values when writing test cases. Compared to that, I think it would be easier to always have one space between each pair of adjacent command values.